pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManagerTest.java

```diff
@@ -173,24 +172,18 @@ private void addAndRemoveNewInstanceConfig(ZkClient zkClient)
     ZNRecord znRecord = new ZNRecord(instanceName);
     zkClient.createPersistent(instanceConfigPath, znRecord);
 
-    List<String> latestAllInstances = _helixResourceManager.getAllInstances();
-    long maxTime = System.currentTimeMillis() + MAX_TIMEOUT_IN_MILLISECOND;
-    while (!latestAllInstances.contains(instanceName) && System.currentTimeMillis() < maxTime) {
-      Thread.sleep(100L);
-      latestAllInstances = _helixResourceManager.getAllInstances();
-    }
-    Assert.assertTrue(System.currentTimeMillis() < maxTime, "Timeout when waiting for adding instance config");
+    TestUtils.waitForCondition(aVoid -> {
```

Comment:
We don't need to wait for condition right?

Reply:
It's possible that direct data accessor still fails to get the instance config because of some other issues like zk connection. Let's still keep the wait for condition here.